cpu/cortexm_common/include/cpu.h

```diff
@@ -58,16 +58,80 @@ extern "C" {
  */
 #define PROVIDES_PM_SET_LOWEST
 
+/**
+ * @name   Pattern to write into the co-processor Access Control Register to
+ *         allow full FPU access
+ *
+ * Used in the `cortexm_init_fpu` inline function below.
+ */
+#define FULL_FPU_ACCESS         (0x00f00000)
+
 /**
  * @brief   Initialization of the CPU
  */
 void cpu_init(void);
 
 /**
  * @brief   Initialize Cortex-M specific core parts of the CPU
+ *
+ * `cortexm_init` calls, in a default order, `cortexm_init_fpu`,
+ * `cortexm_init_isr_priorities`, and `cortexm_init_misc`.  Also
+ * performs other default initialisations, including ones which you
+ * may or may not want.
+ *
+ * Unless you have special requirements (like nRF52 with SD has), it
+ * is sufficient to call just `cortexm_init` and the `cortexm_init_*`
+ * functions do not need to (and should not) be called separately.
+ * If you have conflicting requirements, you may want to have a look
+ * `cpu/nrft/cpu.c` for an example of a non-default approach.
  */
 void cortexm_init(void);
 
+/**
+ * @brief   Initialize Cortex-M FPU
+ *
+ * Called from `cpu/nrf52/cpu.c`, since it cannot use the
+ * whole `cortexm_init` due to conflicting requirements.
+ *
+ * Defined here as a static inline function to allow all
+ * callers to optimise this away if the FPU is not used.
+ */
+static inline void cortexm_init_fpu(void)
+{
+    /* initialize the FPU on Cortex-M4F CPUs */
+#if defined(CPU_ARCH_CORTEX_M4F) || defined(CPU_ARCH_CORTEX_M7)
+    /* give full access to the FPU */
+    SCB->CPACR |= (uint32_t)FULL_FPU_ACCESS;
+#endif
+}
+
+
+#if defined(CPU_CORTEXM_INIT_SUBFUNCTIONS) || defined(doxygen)
+
+/**
+ * @brief   Initialize Cortex-M interrupt priorities
+ *
+ * Called from `cpu/nrf52/cpu.c`, since it cannot use the
+ * whole `cortexm_init` due to conflicting requirements.
+ *
+ * Define `CPU_CORTEXM_INIT_SUBFUNCTIONS` to make this function
+ * available.
+ */
+void cortexm_init_isr_priorities(void);
+
+/**
+ * @brief   Initialize Cortex-M misc functions
+ *
+ * Called from `cpu/nrf52/cpu.c`, since it cannot use the
+ * whole `cortexm_init` due to conflicting requirements.
```

Comment:
editorial comment, not a blocker: If you use `@ref cortexm_init` you will get a clickable link in the generated docs. (or simply `cortexm_init` without backticks, though that relies on Doxygen actually detecting the symbol name in the text)